CMSgov/mdct-mcr
# MDCT-475 - Report context (sneaky hydration(but not actually sneaky; just practical))

## Description
[MDCT-475](https://qmacbis.atlassian.net/browse/MDCT-475) 

What's Report Context? How about the ultimate key to MCR's liberation to live a most humid life!

What do? 
- Hydrate form fields using db data and data that's filled in during that session

How do?
- Using React's Context API we've wrapped the form in a warm blanket of understanding

Why do?
- Makes the user experience much more pleasant when one can tell the values they've entered have gone somewhere (like right in front of them!)

SEE COMMENTS FOR MORE INFO

### How to test
- `./dev local`
- Navigate to `/mcpar/program-information/point-of-contact`
- Start filling out fields and clicking through the pages (using `Save & continue`)
- Once you've filled out a few, go back and review your pages; see if the data is there
- Try changing your response and checking again

## Code author checklist

## Reviewer checklist (two different people)
